### PR TITLE
Do not corrupt custom module set all-xccdf.xml

### LINKED ISSUE #247: Parameter -c changes all-xccdf.xml
Main problem is that if PA run with --contents parameter which takes path to all-xccdf.xml, then this file is changed even though it should not be. When you run PA for a second time it throws the traceback.
``` python
Traceback (most recent call last):
  File "/usr/bin/preupg", line 35, in <module>
    sys.exit(main())
  File "/usr/bin/preupg", line 21, in main
    ret = app.run()
  File "/usr/lib/python2.6/site-packages/preupg/application.py", line 748, in run
    retval = self.scan_system()
  File "/usr/lib/python2.6/site-packages/preupg/application.py", line 507, in scan_system
    self.run_scan_process()
  File "/usr/lib/python2.6/site-packages/preupg/application.py", line 260, in run_scan_process
    self.run_scan(function=self.scanning_progress.show_progress)
  File "/usr/lib/python2.6/site-packages/preupg/application.py", line 276, in run_scan
    return ProcessHelper.run_subprocess(cmd, print_output=False, function=function)
  File "/usr/lib/python2.6/site-packages/preupg/utils.py", line 374, in run_subprocess
    function(stdout_data.decode(settings.defenc))
  File "/usr/lib/python2.6/site-packages/preupg/scanning.py", line 101, in show_progress
    self.output_data.append(u'{0}:{1}'.format(self.names[xccdf_rule], stdout_data.strip()))
```

### PR DESCRIPTION
**Pull request by bocekm**
Preupgrade Assistant was corrupting module set _all-xccdf.xml_ in case the module set path was specified through the _--contents_ option.
Resolves rhbz#[1368823](https://bugzilla.redhat.com/show_bug.cgi?id=1368823)
Resolves #247

### CODE REVIEW

**Comment by bocekm** [comment]
Related issue: https://github.com/upgrades-migrations/preupgrade-assistant/issues/247

**Comment by bocekm** [comment]
@MajoDurco, This PR may seem to become obsolete when https://github.com/upgrades-migrations/preupgrade-assistant/pull/278 gets merged, but it is not - removing the line https://github.com/upgrades-migrations/preupgrade-assistant/pull/281/files#diff-cd214281061a49f12a09f465b45bc87bL463 is key for this PR.

**Comment by pirat89** [comment]
--test--

**Comment by pirat89** [comment]
--test--

**Comment by pirat89** [comment]
--test--

**Comment by pirat89** [comment]
--test--

**Comment by pirat89** [comment]
--test--

**Comment by pirat89** [comment]
--test--

**Comment by pirat89** [comment]
--test--

**Comment by bocekm** [comment]
Rebased with master. Now the code change is simple.

**Comment by AloisMahdal** [comment]
qa_ack: Already covered in downstream test suite.

...just disabled, so just look for the bugzilla number and you will find the relevant part.

**Comment by AloisMahdal** [comment]
I just tried to test with downstream test, where a mock module is created by preupg-content-creator.  I updated the code (to enter '6' and '7' as src/dst versions) but both this PR and master fail to create and execute mock module, though.

preupgrade-assistant-2.4.4-1.201709051242Z.master.el6_8:

    [Errno 2] No such file or directory: u'/root/preupgrade/RHEL6_7/common/release_version

preupgrade-assistant-2.4.4-0.201708291711Z.PR281.el6_8:

    properties.ini file not found inside /usr/share/preupgrade/RHEL6_7 directory

This does not seem to be caused by this PR, so filed separately as #312.

**Comment by bocekm** [comment]
The missing `/root/preupgrade/RHEL6_7/common/release_version` error is related to known issue https://github.com/upgrades-migrations/preupgrade-assistant/issues/279. Currently, as a workaround, you need to have the _preupgrade-assistant-el6toel7_ and _preupgrade-assistant-el6toel7-data_ packages installed, even though you want to run just your custom module set using -c option.

**Comment by AloisMahdal** [comment]
Thanks, the master does not concern me that much though, I just mentioned it hope to provide more context.  The other error is the blocker here.  As you mentioned in #312, it could have already been fixed by #297, but not included in this build; so maybe just rebase this branch on master?  (I tried to do that locally and it did rebase cleanly, I just can't push and trigger the build.)

**Comment by bocekm** [comment]
@AloisMahdal, yeah, I wanted to rebase it but kind of forgot. Sorry. Now it's rebased so hopefully it shouldn't have problems with finding the _properties.ini_.

**Comment by AloisMahdal** [comment]
TL;DR: blocked by #314.

Status update: Current master suffers from the error mentioned above: preupg crashes on non-existence of the releaee_version file, *even* though we're using custom module set.

This problem stems from the fact that downstream build applies  crossarch_migration_support_check.patch, which alters common.py in a way that it relies on release_version file from data package (something entirely specific to el6toel7). This was never supposed to affect custom module sets, the fact that is does is simply a design bug.
 
I've submitted #314 (accompanied with upgrades-migrations/preupgrade-assistant-modules#12, which replaces this method with another one that does not affect custom module sets.

**Comment by bocekm** [comment]
@AloisMahdal, PR rebased, so it includes your PR #314.

**Comment by AloisMahdal** [comment]
Tested with preupgrade-assistant-2.4.4-0.201709191855Z.PR281.el6_8

After building a module set with several mock modules, preupg-xccdf-compose fails:

    Encountered a parse error in /root/RHEL6_7-raw-backup-results/preupgrade_assistant_fupath__builtin/group.xml file, details: Encountered a parse error in file /root/RHEL6_7-raw-backup-results/preupgrade_assistant_fupath__builtin/not_applicable/group.xml.
    Details: junk after document element: line 51, column 0
    No handlers could be found for logger "preupgrade-assistant-debug"

[RHEL6_7-raw.tar.gz](https://github.com/upgrades-migrations/preupgrade-assistant/files/1316220/RHEL6_7-raw.tar.gz)

**Comment by MajoDurco** [comment]
There is an issue for that #310.

**Comment by AloisMahdal** [comment]
Strange enough, this version of preupg-content-creator actually creates old-style modules where the name is not module.ini but (name).ini; I thought that is thing of past.  With combination of mock generator from test library, this caused 2 inis to be included every time.

Anyway, I tested this manually (same build as above) and it looks fine.  Module is not modified by running preupg.  Thanks!

**Comment by pirat89** [comment]
@AloisMahdal it is not about this version. This behaviour you can see even inside the master already. It seems this is still same behaviour since of creation of the tool. It is fixed in PR https://github.com/upgrades-migrations/preupgrade-assistant/pull/311
